pkg/crc/constants/constants.go

```diff
@@ -61,24 +61,31 @@ func GetAdminHelperURL() string {
 	return GetAdminHelperURLForOs(runtime.GOOS)
 }
 
-func defaultBundleForOs(bundleVersion string) map[string]string {
+func defaultBundleForOs(bundleVersion, podmanVersion string, preset crcpreset.Preset) map[string]string {
```

Comment:
I don't think you need to pass `bundleVersion`/`podmanVersion`, the preset is going to be enough since you have a switch on the preset value anyway.
Have you explored adding a
```
type PresetConstants  interface {
    GetDefaultBundle() string
    GetDefaultBundlePath() string
    GetMemory() int
    GetCPUs() int
}

func newPresetConstants(preset preset.Preset) PresetConstants
```
I wonder if this would help makes things slightly cleaner by not having to pass a 'preset' arg to everything, and by avoiding some `if podman ... else ...`. But maybe this won't make much difference, I haven't tried it :)

Reply:
> preset value doesn't contain the required version as part of switch so this is still required.

But `defaultBundleForOs(bundleVersion, podmanVersion string, preset crcpreset.Preset)` is not a nice API at all, depending on the value of `preset`, either `bundleVersion` or `podmanVersion` is going to be ignored, `test/e2e/crcsuite/crcsuite.go ` even ends up not needing one of the args, ...
One way to clean it up is to remove `GetBundleForOs`/`GetDefaultBundleForOs`, which allows to make the whole API simpler. |crcsuite.go` should be able to directly use `bundleName = constants.GetDefaultBundle(preset.OpenShift)`

Reply:
@cfergeau Sure but some follow up commit.

Reply:
Feels like a missed opportunity to first make the code more complicated/harder to review/.., and then on a second pass clean it up, when it can be first cleaned up, and then we never add this hard to follow code.